### PR TITLE
Add SqueezeNet Fire Module 

### PR DESCRIPTION
**Pull request by syed-nazmus-sakib**
## Description
This PR implements **Problem 190: SqueezeNet Fire Module**, which teaches how to use 1x1 convolutions ("squeeze" layer) to reduce dimensionality before applying expensive 3x3 convolutions ("expand" layer).

## Changes
- Added `questions/190_implement-squeezenet-fire-module/` with all required files.
- Includes `meta.json`, `description.md`, `learn.md`, `starter_code.py`, `solution.py`, `example.json`, and `tests.json`.
- (Redundant) Includes `utils/build_bundle.py` fix for Windows encoding to ensure this branch passes CI independently.
- Validated with `utils/validate_questions.py`.

## Checklist
- [x] Question follows template structure
- [x] Solution passes all tests
- [x] Build script runs successfully

### CODE REVIEW

File: questions/190_implement-squeezenet-fire-module/example.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "input": "imput_tensor: (H=32, W=32, C_in=3)\nSqueeze 1x1: s1x1=16 filters\nExpand 1x1: e1x1=64 filters\nExpand 3x3: e3x3=64 filters",
```
Comment (moe18) [inline questions/190_implement-squeezenet-fire-module/example.json]: should be input_tensor (small spelling issue)

File: questions/190_implement-squeezenet-fire-module/tests.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "expected_output": "1.0"
+  },
+  {
+    "test": "import numpy as np\n# Padding test for 3x3\ninput = np.ones((3, 3, 1))\n# Squeeze: weight 1 -> Output 1s\ns_w = np.ones((1, 1, 1, 1))\ns_b = np.zeros(1)\n# Expand 1x1: weights 0 (ignore)\ne1_w = np.zeros((1, 1, 1, 1))\ne1_b = np.zeros(1)\n# Expand 3x3: weight 1 -> Sum neighbors\n# At corner (0,0), with 'same' padding (zeros), only 4 neighbors are non-zero (1s)\n# Sum = 4 * 1 = 4\ne3_w = np.ones((3, 3, 1, 1))\ne3_b = np.zeros(1)\nres = fire_module_forward(input, s_w, s_b, e1_w, e1_b, e3_w, e3_b)\n# Check second channel (Expand 3x3) at corner (0,0)\nprint(res[0, 0, 1])",
+    "expected_output": "4.0"
+  },
+  {
+    "test": "import numpy as np\n# Middle pixel full sum\n# Same setup as above, check middle pixel (1,1)\n# 9 neighbors are 1s -> Sum = 9\ninput = np.ones((3, 3, 1))\ns_w = np.ones((1, 1, 1, 1))\ns_b = np.zeros(1)\ne1_w = np.zeros((1, 1, 1, 1))\ne1_b = np.zeros(1)\ne3_w = np.ones((3, 3, 1, 1))\ne3_b = np.zeros(1)\nres = fire_module_forward(input, s_w, s_b, e1_w, e1_b, e3_w, e3_b)\nprint(res[1, 1, 1])",
+    "expected_output": "9.0"
+  }
```
Comment (moe18) [inline questions/190_implement-squeezenet-fire-module/tests.json:25]: add a test for Non-Zero Bias

File: questions/190_implement-squeezenet-fire-module/tests.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    "expected_output": "1.0"
+  },
+  {
+    "test": "import numpy as np\n# Padding test for 3x3\ninput = np.ones((3, 3, 1))\n# Squeeze: weight 1 -> Output 1s\ns_w = np.ones((1, 1, 1, 1))\ns_b = np.zeros(1)\n# Expand 1x1: weights 0 (ignore)\ne1_w = np.zeros((1, 1, 1, 1))\ne1_b = np.zeros(1)\n# Expand 3x3: weight 1 -> Sum neighbors\n# At corner (0,0), with 'same' padding (zeros), only 4 neighbors are non-zero (1s)\n# Sum = 4 * 1 = 4\ne3_w = np.ones((3, 3, 1, 1))\ne3_b = np.zeros(1)\nres = fire_module_forward(input, s_w, s_b, e1_w, e1_b, e3_w, e3_b)\n# Check second channel (Expand 3x3) at corner (0,0)\nprint(res[0, 0, 1])",
+    "expected_output": "4.0"
+  },
+  {
+    "test": "import numpy as np\n# Middle pixel full sum\n# Same setup as above, check middle pixel (1,1)\n# 9 neighbors are 1s -> Sum = 9\ninput = np.ones((3, 3, 1))\ns_w = np.ones((1, 1, 1, 1))\ns_b = np.zeros(1)\ne1_w = np.zeros((1, 1, 1, 1))\ne1_b = np.zeros(1)\ne3_w = np.ones((3, 3, 1, 1))\ne3_b = np.zeros(1)\nres = fire_module_forward(input, s_w, s_b, e1_w, e1_b, e3_w, e3_b)\nprint(res[1, 1, 1])",
+    "expected_output": "9.0"
```
Comment (moe18) [inline questions/190_implement-squeezenet-fire-module/tests.json:16]: ReLU Is Never Actually Tested in any of the test cases

**Review by moe18 — COMMENTED** [review]
looks good, had a few ideas for the test cases

File: questions/190_implement-squeezenet-fire-module/tests.json (hunk 3 above)
Comment (syed-nazmus-sakib) [reply questions/190_implement-squeezenet-fire-module/tests.json:16]: Thanks for the review! I've pushed the updates. imput_tensor is now input_tensor, and I've added non-zero bias and ReLU edge-case tests to test.json file.
